### ISSUE TITLE
Encountered " ">"

### ISSUE THREAD

**Issue by GhiathNajjar** [javacc/javacc#192]
I'm trying to write some Java code in the jj file. However, it seems that the javacc parser is having trouble reading a block of code.

This is the code:
```

class SymbolTable1 {
    HashMap<Identifier, Tuple<Short, Integer>> hashMap = new HashMap<Identifier, Tuple<Short, Integer>>();

    void addConstant(Identifier id) throws SymbolAlreadyDefinedException1 {
            if (hashMap.containsKey(id))
                throw new SymbolAlreadyDefinedException1("Der Wert existiert!");
            hashMap.put(id, new Tuple<Short, Integer>(id.value, null));
    }

    void addVariable(Identifier id) throws SymbolAlreadyDefinedException1 {
               if (hashMap.containsKey(id))
                   throw new SymbolAlreadyDefinedException1("Der Wert existiert!");
               hashMap.put(id, new Tuple<Short, Integer>(id.address, id.value));
           }

   String getSymbol(String id) throws UnknownSymbolException1 {
           if (hashMap.containsKey(id) == false){
               throw new UnknownSymbolException1("Das Key existiert nicht!");}
           return hashMap.entrySet().stream().filter(i -> i.getKey().id.equals(id)).findFirst().get().getKey().id; // at this line
       }
}

```
If I run javacc then i get the following error:

org.javacc.parser.ParseException: Encountered " ">" "> "" at line 99, column 57.
Was expecting one of:
    "LOOKAHEAD" ...
    "IGNORE_CASE" ...
    "PARSER_BEGIN" ...
    "PARSER_END" ...
    "JAVACODE" ...

That's because of this line code here
`return hashMap.entrySet().stream().filter(i -> i.getKey().id.equals(id)).findFirst().get().getKey().id;`

If I write return "..."; instead, it will work.

So for some reason the javacc Parser cannot read the ">" in that line code

**Comment by zosrothko** [comment]
Can you display what version of JavaCC are you using, the optionblock, your grammar or at least where is the 99 line?

**Comment by MarcMazas** [comment]
Hi. It looks you are using function / lambda related syntax (... -> ...) which is part of Java 8, which is not supported by the currrent versions of JavaCC. A workaround is to put all this kind of code in a helper class outside the grammar file.

**Comment by GhiathNajjar** [comment]
> 
> 
> Can you display what version of JavaCC are you using, the optionblock, your grammar or at least where is the 99 line?

I'm using version 7.0.9 and the 99 line is this line as I mentioned:

`return hashMap.entrySet().stream().filter(i -> i.getKey().id.equals(id)).findFirst().get().getKey().id;`

**Comment by kaikalur** [comment]
Like I mentioned many times - I'm more and more convinced that code should live outside of the jj file. Just use the subclass/superlas mechanisms to do this. With JavaCC 8, we will completely stop checking the code so it should get better. But even then, it's best to keep all code outside of the grammar.

**Comment by zosrothko** [comment]
@GhiathNajjar Since this issue will not be fixed in release 7.0, could you apply the workaround. Thanks
I am closing this issue since no fix will be provided.
